GameFighter/Models/Person.cs

```diff
@@ -16,5 +16,7 @@ public virtual void GetAttack(Warrior warrior)
         {
             Health -= warrior.Attack;
         }
+
+        abstract public void CreateNewModel(ref Warrior unit);
```

Comment:
I don't see any reason to use _ref_ keyword here. Remember generic type parameters and their constraints, or activator class